src/autogen.ml

```diff
@@ -0,0 +1,107 @@
+let mk_input dir input =
+  Filename.concat dir input
+
+let mk_mapper file =
+  let mapper_name = "mapper_" ^ file ^ ".native" in
+  let path_to_bin = Filename.dirname Sys.argv.(0) in
+  Filename.concat path_to_bin mapper_name
+
+let mk_output dir file =
+  Filename.concat dir (file ^ ".ml")
+
+(* template.ml is what the student is going to write into. Therefore, we want
+ * to modify a bit the output of the pretty-printer to add a white line between
+ * each function of the exercise. We also remove trailing whitespaces. *)
+let change_template_fill output template_fill =
+  if Sys.command ("sed -Ei 's/[ \\t]+$//' " ^ output) <> 0 then
+    Printf.eprintf "Could not remove trailing whitespaces in file %s." output;
+  let basename = Filename.basename output in
+  if basename = "template.ml" then begin
+    let default_fill = "Replace this string by your implementation." in
+    let sed =
+      Printf.sprintf "sed -i 's/%s/%s/; /%s/G' %s"
+      default_fill template_fill template_fill output in
+    if Sys.command sed <> 0 then
```

Comment:
The inline treatment of exit status obfuscates the code. Please introduce "exec ?(fatal=false) cmd error_msg" which executes the command "cmd", produces an error message "error_msg" if it fails and stops the programme if "fatal" is true.